extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -125,19 +128,15 @@ private void initialize() {
             tableInfo = ParquetSchemaReader
                     .parseMetadata(parquetMetadata.getFileMetaData().getKeyValueMetaData())
                     .orElse(TableInfo.builder().build());
-            groupingColumns = tableInfo.groupingColumnMap();
-            columnTypes = tableInfo.columnTypeMap();
-            sortingColumns = SortColumnInfo.sortColumns(tableInfo.sortingColumns());
-            version = tableInfo.version();
+
+            isInitialized = true;
```

Comment:
Whenever I see a pattern that looks like this, that is, a volatile write in a happens-once block that is _not_ at the end of the block, it is very suspicious. In this case, I see that you are trying to make sure we don't recursive infinitely via a handleUpdate call.

It would be better IMO to move this back to the end of the block, and add a protected method `handleUpdateInternal` that skips calling `initialize`.